### PR TITLE
IBAN validation, remember rejected content

### LINKED ISSUE #7: IBAN is not fully validated
IBANs contain a inbuilt checksum, with a fairly straightforward way to check this. It would be useful if this checksum could be validated at least server-side (and preferably client-side as well), so the treasurer does not receive invalid payment requests.

The validation instructions (in Dutch) can be found [here](https://nl.wikipedia.org/wiki/International_Bank_Account_Number#Valideren).

### PR DESCRIPTION
**Pull request by maartenberg**
Fixes #7

IBAN validation works, all entries with invalid IBAN-numbers (test: NL00TEST0123456789, valid is NL13TEST0123456789) are rejected, and the contents of the form are kept as well (to ease the process of accepting you're wrong) so that's nice.

### CODE REVIEW

**Comment by maartenberg** [comment]
Currently live at https://digidecs.dev.svsticky.nl

**Comment by tomwassenberg** [comment]
Not really related, but maybe you can also add a label for the submit button, if you're polishing stuff anyway ;)

**Comment by maartenberg** [comment]
Good one, "Ik wil mijn geld terug!"?

**Comment by tomwassenberg** [comment]
For brevity I'd say "Geef geld" ;)

**Comment by maartenberg** [comment]
Done

File: index.php [hunk 1]
```diff
@@ -134,9 +134,43 @@ function validateAll($post, $file) {
...
+
+	// Transliterate into numbers, where A = 10, B = 11 .. Z = 35
+	$transliterated = transnumerate(str_split($transposed));
+
+	return bcmod($transliterated, "97") == "1"; // Magic IBAN constants
+}
+
+function refill($field) {
+	if (isset($_POST[$field]))
+		return 'value="'. $_POST[$field] . '"';
```
Comment (duijf) [inline index.php]: This needs to be sanitized

**Review by duijf — CHANGES_REQUESTED** [review]
Didn't look at this too closely, just one thing that definitely needs fixing (I don't know if I did it properly in other places so feel free to shame me about those :man_shrugging:)

File: index.php (hunk 1 above)
Comment (maartenberg) [reply index.php]: `htmlspecialchars`?

Comment (maartenberg) [reply index.php]: Done, nu actief op dev

**Comment by maartenberg** [comment]
@duijf Escaped enough?

**Pull request merged**